packages/expo-cli/src/commands/eas-build/build/action.ts

```diff
@@ -52,100 +53,148 @@ async function buildAction(projectDir: string, options: BuildOptions): Promise<v
   await ensureGitRepoExistsAsync();
   await ensureGitStatusIsCleanAsync();
 
-  const easConfig: EasConfig = await new EasJsonReader(projectDir, platform).readAsync(profile);
-  const ctx = await createBuilderContextAsync(projectDir, easConfig, {
-    platform,
+  const commandCtx = await createCommandContextAsync({
+    commandPlatform,
+    profile,
+    projectDir,
     nonInteractive: options.parent?.nonInteractive,
     skipCredentialsCheck: options?.skipCredentialsCheck,
     skipProjectConfiguration: options?.skipProjectConfiguration,
   });
-  const projectId = await ensureProjectExistsAsync(ctx.user, {
-    accountName: ctx.accountName,
-    projectName: ctx.projectName,
+
+  const projectId = await ensureProjectExistsAsync(commandCtx.user, {
+    accountName: commandCtx.accountName,
+    projectName: commandCtx.projectName,
   });
-  const scheduledBuilds = await startBuildsAsync(ctx, projectId);
+  const scheduledBuilds = await startBuildsAsync(commandCtx, projectId);
```

Comment:
I would prefer to just leave ctx if there is no ambiguity what context is used. In this case `commandCtx` would be used on as part of builder context